### PR TITLE
Add SpreadProperty, version 2018.0.1

### PR DESCRIPTION
**Pull request by Protryon**
Added `SpreadProperty`, see [the spec](https://www.ecma-international.org/ecma-262/9.0/index.html#prod-PropertyDefinition).

### CODE REVIEW

File: spec.idl [hunk 1]
```diff
@@ -331,6 +331,11 @@ interface ShorthandProperty : ObjectProperty {
   attribute IdentifierExpression name;
 };
 
+// `PropertyDefinition :: ... AssignmentExpression`
+interface SpreadProperty : ObjectProperty {
+  attribute AssignmentExpression expression;
```
Comment (michaelficarra) [inline spec.idl]: `attribute Expression expression;`

Comment (Protryon) [reply spec.idl]: The spec says it's an `AssignmentExpression`. Is the meaning different between our AST and the spec for `AssignmentExpression`?

Comment (bakkot) [reply spec.idl]: Our spec omits intermediate parse nodes, which means that there can be an Expression in that position (by putting parentheses around a sequence).

Comment (Protryon) [reply spec.idl]: Fixed

**Review by michaelficarra — APPROVED** [review]

**Pull request merged**